### PR TITLE
lua-language-server: disable tests on darwin

### LINKED ISSUE #347742: Build failure: lua-language-server-3.11.1 x86_64-darwin
### Steps To Reproduce

1. build lua-language-server on x86_64-darwin

### Build log

```
       > FAILED: luamake
       > ditto build/macos/bin/bootstrap luamake 1>/dev/null
       > /bin/sh: ditto: command not found
       > [36/37] Copy bee.lua/bootstrap/main.lua build/macos/bin/main.lua
       > FAILED: build/macos/bin/main.lua
       > ditto bee.lua/bootstrap/main.lua build/macos/bin/main.lua 1>/dev/null
       > /bin/sh: ditto: command not found
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/hfqhaqnbjzhiy01b412jgivvphjriax1-lua-language-server-3.11.1.drv'.
```

### Additional context

Discovered in downstream test https://github.com/nix-community/nixvim/pull/2399

### Notify maintainers

@figsoda @gepbird @sei40kr 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by gepbird**
Fixes https://github.com/NixOS/nixpkgs/issues/347742
Build times out on hydra during tests but only on x86_64-darwin: https://hydra.nixos.org/build/274693787

cc @khaneliman for testing

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by gepbird** [comment]
@ofborg build lua-language-server

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347769`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>luaPackages.llscheck (lua52Packages.llscheck)</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>luaPackages.llscheck (lua52Packages.llscheck)</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>luaPackages.llscheck (lua52Packages.llscheck)</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>luaPackages.llscheck (lua52Packages.llscheck)</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 347769`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>luaPackages.llscheck (lua52Packages.llscheck)</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
    <li>vscode-extensions.sumneko.lua</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 7 packages failed to build:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>luaPackages.llscheck (lua52Packages.llscheck)</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>luaPackages.llscheck (lua52Packages.llscheck)</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 7 packages built:</summary>
  <ul>
    <li>lua-language-server</li>
    <li>lua51Packages.llscheck</li>
    <li>luaPackages.llscheck (lua52Packages.llscheck)</li>
    <li>lua53Packages.llscheck</li>
    <li>lua54Packages.llscheck</li>
    <li>luajitPackages.llscheck</li>
    <li>lunarvim</li>
  </ul>
</details>

**Comment by gepbird** [comment]
@khaneliman Thanks for these report, it's indeed weird. How do you run darwin builds from linux and vice versa, is it just a new version of nixpkgs review, if so how does that work?

Maybe related: https://github.com/NixOS/nixpkgs/pull/319759#pullrequestreview-2118334260
TLDR: the nix builder on darwin can use user installed homebrew packages, so the same nix package build can differ on on multiple darwin machines.

**Comment by khaneliman** [comment]
> @khaneliman Thanks for these report, it's indeed weird. How do you run darwin builds from linux and vice versa, is it just a new version of nixpkgs review, if so how does that work?

Just using remote builders to ssh build from a different machine.

**Comment by gepbird** [comment]
Also it builds on aarch64-linux for me

**Comment by gepbird** [comment]
> Just using remote builders to ssh build from a different machine. I tested with personal laptop and the darwin community builder.

Probably I'm misunderstanding ssh builds, but when you start a review on your darwin machine, it builds fine. However starting a review on your linux machine that ssh-es into your darwin machine, the build fails.

**Comment by khaneliman** [comment]
> Also it builds on aarch64-linux for me

Weird, I can't get it to build on my machine. Seems like it just times out after `Cleaning... 0 files.`.

> > Just using remote builders to ssh build from a different machine. I tested with personal laptop and the darwin community builder.
> 
> Probably I'm misunderstanding ssh building, but when you start a review on your darwin machine, it builds fine. However starting a review on your linux machine that ssh-es into your darwin machine, the build fails.

Yeah, just seems weird that remote builders still get the ditto error after the change. Looks like it's in the actual build phase and not after the rebuild command. I can play around with it some later.

**Comment by khaneliman** [comment]
Updated issue, but I think we can close this. Sorry, for the hassle.

**Comment by gepbird** [comment]
https://github.com/NixOS/nixpkgs/issues/347742#issuecomment-2418101838
